### ISSUE TITLE
SA1119 (Statement should not use unnecessary parenthesis) false positive

### ISSUE THREAD

**Issue by kindermannhubert** [DotNetAnalyzers/StyleCopAnalyzers#3712]
SA1119 is reported in the following code:
```C#
int x = 0, y = 0;
(b ? ref x : ref y) = 1;
```

But code without parenthesis does not compile.
